frontend/src/routes/(authenticated)/project/[project_code]/+page.ts

```diff
@@ -45,22 +45,38 @@ export async function load(event: PageLoadEvent) {
 								name
 							}
 						}
-						changesets {
-							node
-							parents
-							date
-							user
-							desc
-						}
 					}
 				}
 			`),
       { projectCode }
     );
+  const changesetResultStore = client
+    .queryStore(event.fetch,
+      graphql(`
+        query projectChangesets($projectCode: String!) {
+          projectByCode(code: $projectCode) {
+            id
+            code
+            changesets {
+              node
+              parents
+              date
+              user
+              desc
+            }
+          }
+        }
+      `),
+      { projectCode }
+  );
 
   event.depends(`project:${projectCode}`);
   return {
-    project: result.projectByCode,
+    project: projectResult.projectByCode,
+    changesets: derived(changesetResultStore, result => ({
+      loading: result.fetching,
+      changesets: result.data?.projectByCode?.changesets,
```

Comment:
let's not make changesets nullable here. Fallback to an empty array if the API returns null, then we don't have to write anything else checking if the list is null.